gamla/functional.py

```diff
@@ -314,3 +315,32 @@ def update_in(d, keys, func, default=None, factory=dict):
     else:
         inner[k] = func(default)
     return rv
+
+
+@toolz.curry
+def dataclass_replace(
+    attr_name: Text, attr_mutator: Callable[[Any], Any], dataclass_instance
```

Comment:
I don't know how you are using this but most of the times we just replace values (so the mutator would be `gamla.just(x)` ) maybe add a sugar `dataclass_replace_with_value` ? wdyt ?